pysd/py_backend/utils.py

```diff
@@ -409,4 +517,79 @@ def get_value_by_insensitive_key_or_value(key, dict):
         if real_value.lower() == lower_key:
             return real_value
 
-    return None
\ No newline at end of file
+    return None
+
+
+def rearrange(data, dims, coords):
+    """
+    Returns a xarray.DataArray object with the given coords and dims
+
+    Parameters
+    ---------
+      data: float or xarray.DataArray
+        The input data to rearrange.
+      dims: list
+        Ordered list of the dimensions.
+      coords: dict
+        Dictionary of the dimension names as a keys with their values.
+      switch: bool
+        Flag to denote if the dimensions can be switched. Default True,
+        The False is used to rearrange general expressions
+
+    Returns
+    -------
+      xarray.DataArray
+
+    """
+    # subset used coords in general coords will be the subscript_dict
+    coords = {dim: coords[dim] for dim in dims}
+    if isinstance(data, xr.DataArray):
+        if data.shape == tuple(compute_shape(coords, dims)):
+            # Allows switching dimensions names and transpositions
+            return xr.DataArray(data=data.values, coords=coords, dims=dims)
+
+        # The coordinates are expanded or transposed
+        # TODO replace cleaner version for Python 3 (when deprecate Py2)
+        # return xr.DataArray(0, coords, dims)
+        return xr.DataArray(np.zeros(compute_shape(coords, dims)),
+                            coords, dims) + data
+
+    else:
+        # TODO replace cleaner version for Python 3 (when deprecate Py2)
+        # return xr.DataArray(float(data), coords, dims)
+        return xr.DataArray(np.full(compute_shape(coords, dims),
+                                    float(data)), coords, dims)
+
+
+def round(x):
+    """
+    Redefinition of round function with try except
+    """
+    try:
+        return round(x)
+    except Exception:
```

Comment:
- What exceptions are we expecting? Can we make this more precise than the general exception? 
- What user behavior would make this relevant?
- If we can't round, should we raise an error?
- Is this used by the user? Would it be better to put it in `functions.py`?

Reply:
This was added by @julienmalard . It is used in the Delay Fixed. I have changed it to make it more clear and work with xarrays.

Reply:
@julienmalard can you help me with some of the questions above here?

Reply:
Ah! That was an ugly hack I forgot to fix (sorry)! `round` does not work on xarrays and numpy arrays, so I made this to avoid crashing my models. Of course I really should have implemented this for xarray and numpy too! `np.round(x)` should work for both. Would you like me to make a new pull request based on the new `master` to implement this?